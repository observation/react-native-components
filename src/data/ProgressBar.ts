class ProgressBar {
  get isCompleted(): boolean {
    return this.percentage === 100
  }

  get percentage(): number {
    if (this.currentAmount === 0 || this.maximumAmount === 0) return 0

    return Math.min(100 * (this.currentAmount / this.maximumAmount), 100)
  }

  constructor(
    public label: string,
    public currentAmount: number,
    public maximumAmount: number,
    public backgroundColor: string,
  ) {}
}

export default ProgressBar
