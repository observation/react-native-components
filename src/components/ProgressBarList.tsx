import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

import { IconSuccess } from './Icon'
import ProgressBar from '../data/ProgressBar'
import font from '../styles/font'
import theme from '../styles/theme'

type Props = {
  progressBars: ProgressBar[]
  separator: string
}

const ProgressBarList = ({ progressBars, separator }: Props) => {
  const getMarginBottom = (i: number) => (i === progressBars.length - 1 ? 0 : borderRadius)

  // With small percentages (<10%) the gray bar is visible because of the border radius:
  // the growing bar grows faster than the numbersContainer is shifted to the left
  // If the radius is set to 0 for the first X% (for example 50), this is resolved
  const getBorderRadiusForGrowingBar = (progressBar: ProgressBar) => (progressBar.percentage < 50 ? 0 : borderRadius)

  const allProgressbarTextsAreEmpty = progressBars.every((progressBar) => !progressBar.label)

  return (
    <View style={styles.mainContainer}>
      <View style={styles.labelsContainer}>
        {progressBars.map((progressBar, i) => (
          <View
            key={i}
            style={{
              ...styles.labelContainer,

              marginBottom: getMarginBottom(i),
              backgroundColor: progressBar.backgroundColor,
            }}
          >
            <Text style={styles.label} ellipsizeMode="tail" numberOfLines={1}>
              {progressBar.label}
              <View
                // This view is placed here (instead of a rightPadding) to prevent a 1-pixel
                // white border on some android devices
                style={{
                  backgroundColor: progressBar.backgroundColor,
                  width: allProgressbarTextsAreEmpty ? 0 : 5,
                }}
              />
            </Text>
          </View>
        ))}
      </View>

      <View style={styles.barsContainer}>
        {progressBars.map((progressBar: ProgressBar, i) => (
          <View
            key={i}
            style={{
              ...styles.greyBar,
              marginBottom: getMarginBottom(i),
            }}
          >
            <View
              testID="growing-bar"
              style={{
                borderTopRightRadius: getBorderRadiusForGrowingBar(progressBar),
                borderBottomRightRadius: getBorderRadiusForGrowingBar(progressBar),
                backgroundColor: progressBar.backgroundColor,
                width: `${progressBar.percentage}%`,
              }}
            />
            <View>
              <View
                style={{
                  ...styles.numbersContainer,
                  left: `-${progressBar.percentage}%`,
                  backgroundColor: progressBar.backgroundColor,
                }}
              >
                <View style={{ flexDirection: 'row' }}>
                  <Text style={{ ...styles.numerator }}>{progressBar.currentAmount}</Text>
                  <Text style={{ ...styles.denominator }}>{` ${separator} ${progressBar.maximumAmount}`}</Text>
                </View>
              </View>
            </View>
          </View>
        ))}
      </View>
      <View>
        {progressBars.map((progressBar: ProgressBar, i) => (
          <View
            key={i}
            style={{
              ...styles.checkContainer,
              marginBottom: getMarginBottom(i),
            }}
          >
            {progressBar.isCompleted ? (
              <IconSuccess style="solid" color={progressBar.backgroundColor} size={theme.icon.size.small} />
            ) : (
              <IconSuccess color={theme.color.greySemi} size={theme.icon.size.small} />
            )}
          </View>
        ))}
      </View>
    </View>
  )
}

const barHeight = 16
const borderRadius = barHeight / 2

const styles = StyleSheet.create({
  mainContainer: {
    width: '100%',
    flexDirection: 'row',
  },
  labelsContainer: {
    maxWidth: '50%',
  },
  labelContainer: {
    justifyContent: 'center',
    height: barHeight,
    borderTopLeftRadius: borderRadius,
    borderBottomLeftRadius: borderRadius,
    paddingLeft: borderRadius,
  },
  label: {
    ...font.small,
    color: theme.color.white70,
    justifyContent: 'center',
    lineHeight: barHeight,
  },
  barsContainer: {
    flex: 1,
  },
  greyBar: {
    flexDirection: 'row',
    borderTopRightRadius: borderRadius,
    borderBottomRightRadius: borderRadius,
    backgroundColor: theme.color.greySemi,
    width: '100%',
  },
  numbersContainer: {
    justifyContent: 'center',
    height: barHeight,
    borderTopRightRadius: borderRadius,
    borderBottomRightRadius: borderRadius,
    paddingRight: borderRadius,
  },
  denominator: {
    ...font.small,
    color: theme.color.white70,
    lineHeight: barHeight,
  },
  numerator: {
    ...font.smallBold,
    color: theme.color.white,
    lineHeight: barHeight,
  },
  checkContainer: {
    justifyContent: 'center',
    marginLeft: borderRadius,
    height: barHeight,
  },
})

export default ProgressBarList
