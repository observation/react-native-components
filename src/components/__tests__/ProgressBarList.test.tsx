import React from 'react'

import { render } from '@testing-library/react-native'

import ProgressBar from '../../data/ProgressBar'
import ProgressBarList from '../ProgressBarList'

describe('ProgressBarList', () => {
  test('Show bars', () => {
    // GIVEN
    const bars = [
      new ProgressBar('Label', 0, 6, 'red'),
      new ProgressBar('Label', 1, 6, 'orange'),
      new ProgressBar('Label', 2, 6, 'yellow'),
      new ProgressBar('Label', 3, 6, 'green'),
      new ProgressBar('Label', 4, 6, 'blue'),
      new ProgressBar('Label', 5, 6, 'indigo'),
      new ProgressBar('Label', 6, 6, 'violet'),
    ]

    // THEN
    const { toJSON } = render(<ProgressBarList progressBars={bars} />)
    expect(toJSON()).toMatchSnapshot()
  })

  test('Border radius is 0 with % < 50', () => {
    // GIVEN
    const bars = [new ProgressBar('Label', 0, 6, 'red')]

    // THEN
    const { getByTestId } = render(<ProgressBarList progressBars={bars} />)

    const growingBar = getByTestId('growing-bar')
    expect(growingBar.props.style.borderTopRightRadius).toBe(0)
    expect(growingBar.props.style.borderBottomRightRadius).toBe(0)
  })

  test('Border radius is normal with % > 50', () => {
    // GIVEN
    const bars = [new ProgressBar('Label', 60, 6, 'red')]

    // THEN
    const { getByTestId } = render(<ProgressBarList progressBars={bars} />)

    const growingBar = getByTestId('growing-bar')
    expect(growingBar.props.style.borderTopRightRadius).toBe(8)
    expect(growingBar.props.style.borderBottomRightRadius).toBe(8)
  })
})
