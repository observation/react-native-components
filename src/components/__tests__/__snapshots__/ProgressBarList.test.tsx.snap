// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarList Show bars 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "maxWidth": "50%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "red",
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "paddingLeft": 8,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          {
            "color": "#FFFFFFB4",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "justifyContent": "center",
            "lineHeight": 16,
          }
        }
      >
        Label
        <View
          style={
            {
              "backgroundColor": "red",
              "width": 5,
            }
          }
        />
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "orange",
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "paddingLeft": 8,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          {
            "color": "#FFFFFFB4",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "justifyContent": "center",
            "lineHeight": 16,
          }
        }
      >
        Label
        <View
          style={
            {
              "backgroundColor": "orange",
              "width": 5,
            }
          }
        />
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "yellow",
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "paddingLeft": 8,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          {
            "color": "#FFFFFFB4",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "justifyContent": "center",
            "lineHeight": 16,
          }
        }
      >
        Label
        <View
          style={
            {
              "backgroundColor": "yellow",
              "width": 5,
            }
          }
        />
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "green",
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "paddingLeft": 8,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          {
            "color": "#FFFFFFB4",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "justifyContent": "center",
            "lineHeight": 16,
          }
        }
      >
        Label
        <View
          style={
            {
              "backgroundColor": "green",
              "width": 5,
            }
          }
        />
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "blue",
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "paddingLeft": 8,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          {
            "color": "#FFFFFFB4",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "justifyContent": "center",
            "lineHeight": 16,
          }
        }
      >
        Label
        <View
          style={
            {
              "backgroundColor": "blue",
              "width": 5,
            }
          }
        />
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "indigo",
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "paddingLeft": 8,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          {
            "color": "#FFFFFFB4",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "justifyContent": "center",
            "lineHeight": 16,
          }
        }
      >
        Label
        <View
          style={
            {
              "backgroundColor": "indigo",
              "width": 5,
            }
          }
        />
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "violet",
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 0,
          "paddingLeft": 8,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          {
            "color": "#FFFFFFB4",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "justifyContent": "center",
            "lineHeight": 16,
          }
        }
      >
        Label
        <View
          style={
            {
              "backgroundColor": "violet",
              "width": 5,
            }
          }
        />
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E6E6E6",
          "borderBottomRightRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "red",
            "borderBottomRightRadius": 0,
            "borderTopRightRadius": 0,
            "width": "0%",
          }
        }
        testID="growing-bar"
      />
      <View>
        <View
          style={
            {
              "backgroundColor": "red",
              "borderBottomRightRadius": 8,
              "borderTopRightRadius": 8,
              "height": 16,
              "justifyContent": "center",
              "left": "-0%",
              "paddingRight": 8,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 16,
                }
              }
            >
              0
            </Text>
            <Text
              style={
                {
                  "color": "#FFFFFFB4",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 16,
                }
              }
            >
               of 6
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#E6E6E6",
          "borderBottomRightRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "orange",
            "borderBottomRightRadius": 0,
            "borderTopRightRadius": 0,
            "width": "16.666666666666664%",
          }
        }
        testID="growing-bar"
      />
      <View>
        <View
          style={
            {
              "backgroundColor": "orange",
              "borderBottomRightRadius": 8,
              "borderTopRightRadius": 8,
              "height": 16,
              "justifyContent": "center",
              "left": "-16.666666666666664%",
              "paddingRight": 8,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 16,
                }
              }
            >
              1
            </Text>
            <Text
              style={
                {
                  "color": "#FFFFFFB4",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 16,
                }
              }
            >
               of 6
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#E6E6E6",
          "borderBottomRightRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "yellow",
            "borderBottomRightRadius": 0,
            "borderTopRightRadius": 0,
            "width": "33.33333333333333%",
          }
        }
        testID="growing-bar"
      />
      <View>
        <View
          style={
            {
              "backgroundColor": "yellow",
              "borderBottomRightRadius": 8,
              "borderTopRightRadius": 8,
              "height": 16,
              "justifyContent": "center",
              "left": "-33.33333333333333%",
              "paddingRight": 8,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 16,
                }
              }
            >
              2
            </Text>
            <Text
              style={
                {
                  "color": "#FFFFFFB4",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 16,
                }
              }
            >
               of 6
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#E6E6E6",
          "borderBottomRightRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "green",
            "borderBottomRightRadius": 8,
            "borderTopRightRadius": 8,
            "width": "50%",
          }
        }
        testID="growing-bar"
      />
      <View>
        <View
          style={
            {
              "backgroundColor": "green",
              "borderBottomRightRadius": 8,
              "borderTopRightRadius": 8,
              "height": 16,
              "justifyContent": "center",
              "left": "-50%",
              "paddingRight": 8,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 16,
                }
              }
            >
              3
            </Text>
            <Text
              style={
                {
                  "color": "#FFFFFFB4",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 16,
                }
              }
            >
               of 6
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#E6E6E6",
          "borderBottomRightRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "blue",
            "borderBottomRightRadius": 8,
            "borderTopRightRadius": 8,
            "width": "66.66666666666666%",
          }
        }
        testID="growing-bar"
      />
      <View>
        <View
          style={
            {
              "backgroundColor": "blue",
              "borderBottomRightRadius": 8,
              "borderTopRightRadius": 8,
              "height": 16,
              "justifyContent": "center",
              "left": "-66.66666666666666%",
              "paddingRight": 8,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 16,
                }
              }
            >
              4
            </Text>
            <Text
              style={
                {
                  "color": "#FFFFFFB4",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 16,
                }
              }
            >
               of 6
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#E6E6E6",
          "borderBottomRightRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "indigo",
            "borderBottomRightRadius": 8,
            "borderTopRightRadius": 8,
            "width": "83.33333333333334%",
          }
        }
        testID="growing-bar"
      />
      <View>
        <View
          style={
            {
              "backgroundColor": "indigo",
              "borderBottomRightRadius": 8,
              "borderTopRightRadius": 8,
              "height": 16,
              "justifyContent": "center",
              "left": "-83.33333333333334%",
              "paddingRight": 8,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 16,
                }
              }
            >
              5
            </Text>
            <Text
              style={
                {
                  "color": "#FFFFFFB4",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 16,
                }
              }
            >
               of 6
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#E6E6E6",
          "borderBottomRightRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "marginBottom": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "violet",
            "borderBottomRightRadius": 8,
            "borderTopRightRadius": 8,
            "width": "100%",
          }
        }
        testID="growing-bar"
      />
      <View>
        <View
          style={
            {
              "backgroundColor": "violet",
              "borderBottomRightRadius": 8,
              "borderTopRightRadius": 8,
              "height": 16,
              "justifyContent": "center",
              "left": "-100%",
              "paddingRight": 8,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 16,
                }
              }
            >
              6
            </Text>
            <Text
              style={
                {
                  "color": "#FFFFFFB4",
                  "fontFamily": "Ubuntu",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 16,
                }
              }
            >
               of 6
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        {
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "marginLeft": 8,
        }
      }
    >
      <Icon
        color="#E6E6E6"
        name="circle-check"
        size={12}
        type="light"
      />
    </View>
    <View
      style={
        {
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "marginLeft": 8,
        }
      }
    >
      <Icon
        color="#E6E6E6"
        name="circle-check"
        size={12}
        type="light"
      />
    </View>
    <View
      style={
        {
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "marginLeft": 8,
        }
      }
    >
      <Icon
        color="#E6E6E6"
        name="circle-check"
        size={12}
        type="light"
      />
    </View>
    <View
      style={
        {
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "marginLeft": 8,
        }
      }
    >
      <Icon
        color="#E6E6E6"
        name="circle-check"
        size={12}
        type="light"
      />
    </View>
    <View
      style={
        {
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "marginLeft": 8,
        }
      }
    >
      <Icon
        color="#E6E6E6"
        name="circle-check"
        size={12}
        type="light"
      />
    </View>
    <View
      style={
        {
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 8,
          "marginLeft": 8,
        }
      }
    >
      <Icon
        color="#E6E6E6"
        name="circle-check"
        size={12}
        type="light"
      />
    </View>
    <View
      style={
        {
          "height": 16,
          "justifyContent": "center",
          "marginBottom": 0,
          "marginLeft": 8,
        }
      }
    >
      <Icon
        color="violet"
        name="circle-check"
        size={12}
        type="solid"
      />
    </View>
  </View>
</View>
`;
